lib/CodeBuilder/Adapter/TolerantParser/Updater/AbstractMethodUpdater.php

```diff
@@ -141,19 +146,21 @@ private function updateOrAddParameters(Edits $edits, Parameters $parameters, Met
 
         $renderedParameters = [];
 
+        /** @var ParameterDeclarationList|null $existingParameterDeclaration */
+        $existingParameterDeclaration = $methodDeclaration->parameters;
```

Comment:
upstream TP has wrong declaration here.

Reply:
Exactly, can we change that somehow?

Reply:
It would be fixing a needle in a haystack of wrong types. I've started pushing PHPStan fixes up stram: https://github.com/microsoft/tolerant-php-parser/pull/396

but upstream is sometimes slow, and I wanted to add some changes which I didn't have the capacity to justify, so we use a fork: https://github.com/phpactor/tolerant-php-parser

Reply:
I'll make a note in the other package. Should I also fix it here?

Reply:
i would rather we updated all PHPStan level than try and fix each individual error. so fine to leave this.